server/controller/root.py

```diff
@@ -106,6 +110,63 @@ def historical():
     }
 
 
+@get('/upload_logs')
+@kajiki_view('upload_logs')
+@authenticated()
+def upload_logs():
+    api_key = config.user_management_interface.get_api_key_for_user(request)
+    return {
+        'page': 'upload_logs',
+        'raw_logs': '',
+    }
+
+
+@post('/upload_logs')
+@kajiki_view('upload_logs')
+@authenticated()
+def upload_logs():
+    api_key = config.user_management_interface.get_api_key_for_user(request)
+
+    os_type = request.forms.get('os_type')
+    upload = request.files.get('upload')
+    pasted = request.forms.get('message')
+    return_val = {
+        'page': 'upload_logs',
+        'raw_logs': request.forms.get('message'),
+        'log_entries': Markup(''),
+    }
+
+    if upload is not None:
+        message = str(upload.file.read(), 'utf-8')
+    elif not pasted.isspace() and pasted != '':
+        message = pasted
+    else:
+        return_val['flash'] = {
+            'content': 'Please upload a file or paste logs.',
+            'cls': 'error',
+        }
+        return return_val
+
+    try:
+        parsed_message = parsing_lib.LogParser.parse(message, os_type)
+        log_entries = parsing_lib.LogParser.convert_to_html(parsed_message)
+        return_val['log_entries'] = Markup(log_entries)
+    except:
+        e = sys.exc_info()[0]
+        return_val['flash'] = {
+            'content': 'Log format error: %s' % str(e),
+            'cls': 'error',
+        }
+    finally:
+        return return_val
+
+@route('/new_login')
+@kajiki_view('new_login')
+def new_login():
+    """Shows new login page."""
+    return {'page': 'new_login'}
```

Comment:
@amandagiles95, remove this. I think we accidentally added this back in during the merge.